src/commands/react-native/xcode.ts

```diff
@@ -0,0 +1,287 @@
+// tslint:disable: no-null-keyword
+import {spawn} from 'child_process'
+import {Cli, Command} from 'clipanion'
+import {existsSync} from 'fs'
+import {UploadCommand} from './upload'
+
+export class XCodeCommand extends Command {
+  public static usage = Command.Usage({
+    description: 'Bundle React Native code and images in XCode and send sourcemaps to Datadog.',
+    details: `
+            This command will bundle the react native code and images and then upload React Native sourcemaps and their corresponding javascript bundle to Datadog in order to un-minify front-end stack traces received by Datadog.
+            See README for details.
+        `,
+    examples: [
+      [
+        'Usage as XCode build phase for RN < 0.69',
+        'set -e\nexport SOURCEMAP_FILE=./build/main.jsbundle.map\nexport NODE_BINARY=node\n# Replace /opt/homebrew/bin/node (resp. /opt/homebrew/bin/yarn) by the value of $(command -v node) (resp. $(command -v yarn))\n/opt/homebrew/bin/node /opt/homebrew/bin/yarn datadog-ci react-native xcode node_modules/react-native/scripts/react-native-xcode.sh\n',
+      ],
+      [
+        'Usage as XCode build phase for RN >= 0.69',
+        'set -e\nexport SOURCEMAP_FILE=./main.jsbundle.map\nWITH_ENVIRONMENT="../node_modules/react-native/scripts/xcode/with-environment.sh"\nREACT_NATIVE_XCODE="./datadog-sourcemaps.sh"\n\n/bin/sh -c "$WITH_ENVIRONMENT $REACT_NATIVE_XCODE"\n',
+      ],
+    ],
+  })
+
+  private composeSourcemapsPath = 'node_modules/react-native/scripts/compose-source-maps.js'
+  private dryRun = false
+  private force = false
+  private scriptPath = 'node_modules/react-native/scripts/react-native-xcode.sh'
```

Comment:
If you want to support global installation of datadog-ci, I'm not certain the relative paths will work. It might be necessary to use something like `__dirname` to build the path

Reply:
Fair point, I can also make it a bit more robust for workspaces, for instance by using `require.resolve('react-native')`. Let me investigate on this